Framework reference restrictions for plugins (#15701)
dotnet/docs: docs/core/tutorials/creating-app-with-plugin-support.md

## Plugin target framework recommendations

Because plugin dependency loading uses the *.deps.json* file, there is a gotcha related to the plugin's target framework. Specifically, your plugins should target a runtime, such as .NET Core 3.0, instead of a version of .NET Standard. The *.deps.json* file is generated based on which framework the project targets, and since many .NET Standard-compatible packages ship reference assemblies for building against .NET Standard and implementation assemblies for specific runtimes, the *.deps.json* may not correctly see implementation assemblies, or it may grab the .NET Standard version of an assembly instead of the .NET Core version you expect.

## Plugin framework references

> Currently, plugins can't introduce new frameworks into the process. So for example, you can't load a plugin that uses the `Microsoft.AspNetCore.App` framework into an application that only uses the root `Microsoft.NETCore.App` framework. The host application must declare references to all frameworks needed by plugins.

Reviewer: ```suggestion
Currently, plugins can't introduce new frameworks into the process. For example, you can't load a plugin that uses the `Microsoft.AspNetCore.App` framework into an application that only uses the root `Microsoft.NETCore.App` framework. The host application must declare references to all frameworks needed by plugins.
```

> Currently plugins can not introduce new frameworks into the process. So for example trying to load a plugin which uses ASP.NET (`Microsoft.AspNetCore.App` framework) into an application which only uses the root `Microsoft.NETCore.App` framework is not allowed. The host application must declare references to all frameworks it itself and all potential plugins might need.

Reviewer 2: ```suggestion
Currently, plugins can't introduce new frameworks into the process. So for example, you can't load a plugin that uses the `Microsoft.AspNetCore.App` framework into an application that only uses the root `Microsoft.NETCore.App` framework. The host application must declare references to all frameworks needed by plugins.
```
